pkg/server/server.go

```diff
@@ -383,31 +386,32 @@ func (s *Server) VerifyEmail(c *gin.Context, redirectURL string) {
 	if !exists {
 		s.redirectError(c, errors.ErrInvalidAccessToken)
 		// Redirecting user to UI if the user is not authorized
-		c.Redirect(http.StatusUnauthorized, redirectURL)
+		c.Redirect(http.StatusPermanentRedirect, redirectURL)
 		return
 	}
 	jwtUserCredentials := jwtUser.(*models.UserCredentials)
 
 	if jwtUserCredentials.UnverifiedEmail != nil {
-		*jwtUserCredentials.Email = jwtUserCredentials.GetUnverifiedEmail()
-		*jwtUserCredentials.UnverifiedEmail = ""
+		tmp := jwtUserCredentials.GetUnverifiedEmail()
+		jwtUserCredentials.Email = &tmp
+		jwtUserCredentials.UnverifiedEmail = nil
 	} else {
 		log.Errorln("No email found to be verified for user uid: ", jwtUserCredentials.GetUID())
 		c.JSON(http.StatusBadRequest, gin.H{
 			"error": "No email found to be verified",
 		})
 		// Redirecting user to UI if no email found in field `unverified_email`
-		c.Redirect(http.StatusBadRequest, redirectURL)
+		c.Redirect(http.StatusPermanentRedirect, redirectURL)
```

Comment:
Why change the error code?

Reply:
All the error codes are changed for redirection as for redirection only below error codes can be given else it will give error 
```
	StatusMultipleChoices   = 300 // RFC 7231, 6.4.1
	StatusMovedPermanently  = 301 // RFC 7231, 6.4.2
	StatusFound             = 302 // RFC 7231, 6.4.3
	StatusSeeOther          = 303 // RFC 7231, 6.4.4
	StatusNotModified       = 304 // RFC 7232, 4.1
	StatusUseProxy          = 305 // RFC 7231, 6.4.5
	_                       = 306 // RFC 7231, 6.4.6 (Unused)
	StatusTemporaryRedirect = 307 // RFC 7231, 6.4.7
	StatusPermanentRedirect = 308 // RFC 7538, 3

```

Reply:
How will UI know that an error has occurred if we are sending 301 as response code?

Reply:
This status code will be shown to origin of the request that will be a button click on a email. Actual status and code that UI will come across is the above one with `c.JSON`